File: scripts/setup-db.js
const fs = require("node:fs");
const path = require("node:path");
const { Pool } = require("pg");
const dotenv = require("dotenv");
const { VENTURES } = require("../data/venues");

dotenv.config();

async function run() {
  const connectionString = process.env.DATABASE_URL;
  if (!connectionString) {
    throw new Error("DATABASE_URL is missing.");
  }

  const schemaPath = path.join(__dirname, "..", "db", "schema.sql");
  const schemaSql = fs.readFileSync(schemaPath, "utf8");
  const pool = new Pool({
    connectionString,
    ssl: { rejectUnauthorized: false },
  });

  try {
    await pool.query(schemaSql);

    for (const venue of VENTURES) {
      await pool.query(
        `insert into venues (slug, name, tagline, table_prefix, default_table_count)
         values ($1, $2, $3, $4, $5)
         on conflict (slug)
         do update set
           name = excluded.name,
           tagline = excluded.tagline,
           table_prefix = excluded.table_prefix,
           default_table_count = excluded.default_table_count,
           updated_at = now()`,
        [venue.slug, venue.name, venue.tagline, venue.tablePrefix, venue.defaultTableCount],
      );
    }

    console.log(`Database ready. Seeded ${VENTURES.length} ventures.`);
  } finally {
    await pool.end();
  }
}

run().catch((error) => {
  console.error(error.message);
  process.exitCode = 1;
});

File: scripts/setup_db.py
from pathlib import Path
import os
import sys

from dotenv import load_dotenv
import psycopg

PROJECT_ROOT = Path(__file__).resolve().parent.parent
if str(PROJECT_ROOT) not in sys.path:
    sys.path.insert(0, str(PROJECT_ROOT))

from data.venues import VENTURES


def main() -> None:
    load_dotenv()
    database_url = os.getenv("DATABASE_URL")
    if not database_url:
        raise RuntimeError("DATABASE_URL is missing.")

    schema_sql = (PROJECT_ROOT / "db" / "schema.sql").read_text(encoding="utf-8")

    with psycopg.connect(database_url, sslmode="require") as connection:
        with connection.cursor() as cursor:
            cursor.execute(schema_sql)
            for venue in VENTURES:
                cursor.execute(
                    """
                    insert into venues (slug, name, tagline, table_prefix, default_table_count)
                    values (%s, %s, %s, %s, %s)
                    on conflict (slug)
                    do update set
                      name = excluded.name,
                      tagline = excluded.tagline,
                      table_prefix = excluded.table_prefix,
                      default_table_count = excluded.default_table_count,
                      updated_at = now()
                    """,
                    (
                        venue["slug"],
                        venue["name"],
                        venue["tagline"],
                        venue["table_prefix"],
                        venue["default_table_count"],
                    ),
                )

        connection.commit()

    print(f"Database ready. Seeded {len(VENTURES)} ventures.")


if __name__ == "__main__":
    main()
